### PR TITLE
Add FXIOS-5795 [v116] biometrics support for bottom sheet select a card flow

### PR DESCRIPTION
**Pull request by nbhasin2**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5795)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15129)

### Description
Added biometric check before showing list of cards a user can select

https://github.com/mozilla-mobile/firefox-ios/assets/8919439/b5b86d1a-dc04-46bc-9dba-1166c22f9df3

### Pull requests checks where applicable
- [ ] Fill in the three TODOs above (tickets number and description of your work)
- [ ] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1884,15 +1885,37 @@ class BrowserViewController: UIViewController, SearchBarLocationProvider, Themea
...
+                self.showBottomSheetCardViewController(creditCard: nil,
+                                                       decryptedCard: nil,
+                                                       viewType: .selectSavedCard)
+            case .deviceOwnerFailed:
+                break // Keep showing bvc
+            case .passCodeRequired:
+                let passcodeViewController = DevicePasscodeRequiredViewController()
+                passcodeViewController.profile = self.profile
+                self.navigationController?.pushViewController(passcodeViewController,
+                                                              animated: true)
```
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController.swift:1914]: FYI: This can be moved to use coordinators as we have done so in the settings area

**Review by OrlaM — APPROVED** [review]

**Comment by nbhasin2** [comment]
image

Last commit was only to fix conflicts

**Comment by nbhasin2** [comment]
One last re-run as the two failing tests have nothing to do with this change

**Comment by nbhasin2** [comment]
BR is green
image

**Pull request merged**